lsp-serenata.el

```diff
@@ -39,21 +40,98 @@ It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
   :group 'lsp-serenata
   :type 'file)
 
+(defcustom lsp-serenata-uris
+  []
+  "A list of folders to index for your project.
+This does not have to include the root of the project itself, in
+case you have need of an exotic configuration where the root of
+the project is at some location but your actual PHP code is
+somewhere else.  Note that if you are running Serenata in a
+container, you will have to ensure that these URI's are mapped
+inside it.  Avoid using file paths containing spaces. This is
+currently broken due to apparant PHP quirks.  By default, the
+value is taken from the lsp workspace location."
+  :group 'lsp-serenata
+  :type 'lsp-string-vector)
+
+(defcustom lsp-serenata-php-version
+  7.3
+  "Allows you to specify the PHP version your project is written in.
+At the moment this directive is still ignored, but it will
+influence functionality such as refactoring in the future, where
+older PHP versions may not support scalar type hints, which may
+then be omitted from places such as getters and setters."
+  :group 'lsp-serenata
+  :type 'number)
+
+(defcustom lsp-serenata-file-extensions
+  ["php"]
+  "List of file extensions (without dot) to process.
+Files that do not match this whitelist will be ignored during
+indexing.  Usually you'll want to set this to at least include
+php, as it is the most common PHP extension.  phpt is not
+included by default as it is often used to contain test code that
+is not directly part of the code.  Note that for existing
+projects, removing extensions will not not automatically prune
+files having them from the index if they are already present.
+Adding new ones will cause the files having them to be picked up
+on the next project initialization."
+  :group 'lsp-serenata
+  :type 'lsp-string-vector)
+
+(defcustom lsp-serenata-index-database-uri (lsp--path-to-uri (f-join  "~/.emacs.d/index.sqlite"))
```

Comment:
dont you need home directory here?

Reply:
Yes, it still works with GNU/Linux, but it may break in Windows/Mac I don't know.
I'll add the user-emacs-directory variable